client/src/document/baseline-indicator.tsx

```diff
@@ -0,0 +1,89 @@
+import { WebFeatureStatus } from "../../../libs/types/document";
+import { Icon } from "../ui/atoms/icon";
+import "./baseline-indicator.scss";
+
+const ENGINES = [
+  { name: "Blink", browsers: ["Chrome", "Edge"] },
+  { name: "Gecko", browsers: ["Firefox"] },
+  { name: "WebKit", browsers: ["Safari"] },
+];
+
+export function BaselineIndicator({ status }: { status?: WebFeatureStatus }) {
+  const supported = (browser: string) => {
+    const version: string | boolean | undefined =
+      status?.support?.[browser.toLowerCase()];
+    return Boolean(
+      status?.is_baseline || typeof version === "string" || version
+    );
+  };
+
+  return status && typeof status.is_baseline !== "undefined" ? (
+    <details
+      className={`baseline-indicator ${status.is_baseline ? "supported" : ""}`}
+    >
+      <summary>
+        <span
+          className="indicator"
+          role="img"
+          aria-label={status.is_baseline ? "Baseline Check" : "Baseline Cross"}
+        />
+        <h2>
+          Baseline<span>: </span>
+          <span className="not-bold">
+            {status.is_baseline ? "Widely supported" : "Not widely supported"}
+          </span>
+        </h2>
+        <div className="browsers">
+          {ENGINES.map(({ browsers }) => (
+            <span
+              className="engine"
+              title={browsers
+                .map((browser, index, array) => {
+                  const previous =
+                    index > 0 ? supported(array[index - 1]) : undefined;
+                  const current = supported(browser);
+                  return typeof previous === "undefined"
+                    ? current
+                      ? `Supported in ${browser}`
+                      : `Not widely supported in ${browser}`
+                    : current === previous
+                    ? ` and ${browser}`
+                    : current
+                    ? `, and supported in ${browser}`
+                    : `, and not widely supported in ${browser}`;
+                })
+                .join("")}
```

Comment:
🟡 Given its complexity, maybe worth extracting to a function `engineTitle(browsers)`?